batch/batch/server/server.py

```diff
@@ -223,8 +217,8 @@ def _delete_k8s_resources(self):
             del pod_name_job[self._pod_name]
             self._pod_name = None
 
-    def _read_logs(self):
-        logs = {jt.name: _read_file(_log_path(self.id, jt.name))
+    async def _read_logs(self):
+        logs = {jt.name: await blocking_to_async(read_gs_log_file, instance_id, self.id, jt.name)
```

Comment:
I prefer pushing the `blocking_to_async` all the way down to the blocking function or to a library interface (in this case: `client.bucket`, `bucket.blob` and the functions on `f`. This allows other operations to jump in-between these separate blocking calls potentially providing more efficient use of the limited threads in the blocking thread pool.

Reply:
This is tricky to do with how the code is currently setup to not have circular dependencies and how blocking_to_async needs to be in `server.py` so `app` is defined. I'll think about it.

Reply:
`blocking_to_async` should take the thread pool.  Then you can pass the thread pool to `read_gs_log_file` which can call `blocking_to_async`.